api/v1beta1/azuremanagedcontrolplane_default.go

```diff
@@ -73,6 +80,9 @@ func (m *AzureManagedControlPlane) setDefaultSubnet() {
 	}
 	if m.Spec.VirtualNetwork.Subnet.CIDRBlock == "" {
 		m.Spec.VirtualNetwork.Subnet.CIDRBlock = defaultAKSNodeSubnetCIDR
+		if ptr.Deref(m.Spec.NetworkPluginMode, "") == NetworkPluginModeOverlay {
+			m.Spec.VirtualNetwork.Subnet.CIDRBlock = defaultAKSNodeSubnetCIDRForOverlay
```

Comment:
One final validation possibility here... so I see we're assigning an overlay-specific CIDR value for VNET and subnet if the user doesn't declare one. Is there any validation we can add here for the overlay CIDRs? Does AKS require that we use particular nets here?

We're already doing some basic hygiene in `validateSubnetCIDR` and `validateVnetCIDR` (`api/v1beta1/azurecluster_validation.go`) so we get that for free.

Reply:
Are you suggesting to call the same `validateVnetCIDR` and `validateSubnetCIDR` on these values?

> Does AKS require that we use particular nets here?

No AFAICT, these overlay-specific defaults match AKS's defaults when overlay is enabled per https://github.com/kubernetes-sigs/cluster-api-provider-azure/issues/3678#issuecomment-1715680082.

Reply:
I'm saying we're already calling those validate funcs, so that's good (as far as I can see we're reusing existing property CIDR fields w/ this implementation)

My question is what happens when a user *explicitly* passes in `"10.0.0.0/8"` (for example) as the VNET CIDR _and_ enables overlay CNI? Is AKS opinionated: "Hey, that CIDR block is reserved for non-overlay configurations!".

If we don't know all the answers to the above, I think it's fair to keep that out of scope, just curious.

Reply:
> My question is what happens when a user _explicitly_ passes in `"10.0.0.0/8"` (for example) as the VNET CIDR _and_ enables overlay CNI? Is AKS opinionated: "Hey, that CIDR block is reserved for non-overlay configurations!".

The comment I linked above shows the error with the non-overlay defaults, and it wasn't that specific:
`NetworkProfile CIDR 10.244.0.0/16 overlaps with reserved VNET CIDR 10.0.0.0/8`

I don't think a user explicitly passing in the value would affect that since it results in the same config sent to AKS either way.

Reply:
I think we can add more explicit CAPZ validations that mirror AKS validations over time if it has user value (in most cases failing the declarative spec is more user-friendly than looking for an API error in the CAPZ controller logs). Shouldn't block this change.